users/userModel.js

```diff
@@ -0,0 +1,30 @@
+
+const db = require('../database/db-config')
+
+module.exports = {
+    add,
+    findBy,
+    findById,
+    getAllUsers
+}
+
+async function add(user) {
+    const [user_id] = await db('users').insert(user);
+
+    return findById(user_id)
+}
+
+function findById(user_id) {
+    return db('users')
+        .where({ user_id })
+        .first();
+}
+
+function findBy(filter) {
+    return db('users').where(filter);
+}
+
+  function getAllUsers() {
```

Comment:
when getting all users, the password should not be amongst the things selected. thats a security risk

Reply:
...goodness, silly mistake!